examples/quantum_mechanics/photon_to_electron_positron_pair.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+from sympy import solve, Symbol, Eq
+from symplyphysics import print_expression, units, convert_to, quantities
+from symplyphysics.laws.waves import (
+    photon_energy_is_proportional_to_frequency as energy_law,
+    wavelength_from_wave_speed_and_period as wavelength_law,
+)
+from symplyphysics.laws.relativistic import energy_is_mass as rest_mass_law
+from symplyphysics.laws.kinematic import temporal_frequency_from_period as frequency_law
+
+# Description
+## Can a free photon with enough energy turn into an electron-positron pair?
+
+# First of all, the conversion of a photon into an electron-positron pair is only possible
+# in the presence of another particle, otherwise the process would violate the law of conservation
+# of momentum. Next, the more the mass of the third particle is, the less kinetic energy it gets
+# after the conversion, and we can omit its kinetic energy out of the equation.
+
+photon_frequency = Symbol("photon_frequency")
+electron_rest_mass = Symbol("electron_rest_mass")
+
+photon_energy = energy_law.law.rhs.subs(energy_law.photon_frequency, photon_frequency)
+
+electron_rest_energy = rest_mass_law.law.rhs.subs(rest_mass_law.rest_mass, electron_rest_mass)
+positron_rest_energy = electron_rest_energy
+
+# If the electron and positron are at rest after their creation, their total energy would be the lowest
+# out of all possible values and be equal to the rest energy of the system.
+
+electron_positron_pair_rest_energy = electron_rest_energy + positron_rest_energy
+
+# TODO: make law for this
+energy_conservation_eqn = Eq(photon_energy, electron_positron_pair_rest_energy)
+
+photon_wavelength_expr = solve(
+    (
+        energy_conservation_eqn,
+        frequency_law.law.subs({
+            frequency_law.period: wavelength_law.oscillation_period,
+            frequency_law.temporal_frequency: photon_frequency,
+        }),
+        wavelength_law.law.subs(wavelength_law.propagation_speed, units.speed_of_light),
+    ),
+    (
+        photon_frequency,
+        wavelength_law.oscillation_period,
+        wavelength_law.wavelength,
+    ),
+    dict=True,
+)[0][wavelength_law.wavelength]
+
+print(f"Threshold photon wavelength:\n\n{print_expression(photon_wavelength_expr)}\n")
```

Comment:
Not clear, what is threshold. Is it minimal or maximal wavelength to form a pair?

Reply:
Explained it